services/alertmanager/alertmanager.go

```diff
@@ -359,20 +359,33 @@ func (svc *Service) populateConfig(cfg *alertmanager.Config) error {
 		cfg.Global = &alertmanager.GlobalConfig{}
 	}
 
-	// make sure that "empty" receiver is there
-	var emptyFound bool
-	for _, r := range cfg.Receivers {
-		if r.Name == "empty" {
-			emptyFound = true
-			break
+	findReceiver := func(name string) int {
+		for i, r := range cfg.Receivers {
+			if r.Name == name {
+				return i
+			}
 		}
+		return -1
 	}
+
+	emptyFound := findReceiver("empty") != -1
+	// make sure that "empty" receiver is there
 	if !emptyFound {
 		cfg.Receivers = append(cfg.Receivers, &alertmanager.Receiver{
 			Name: "empty",
 		})
 	}
 
+	disabledReceiver := &alertmanager.Receiver{
+		Name: "disabled",
+	}
+	disabledReceiverIndex := findReceiver("disabled")
+	if disabledReceiverIndex >= 0 {
+		cfg.Receivers[disabledReceiverIndex] = disabledReceiver
```

Comment:
why do we update it here?

Reply:
so user can't add his "disabled" receiver by using file.
Same thing as he can add "empty" (default) receive.

Reply:
Yeah, but why do we replace it?

Reply:
Any better solution how to forbid user to add this receiver?
Check after yaml is parsed? And what then?

Reply:
ok, so we let users to override empty, but don't let to override disabled, right?
If it is, then everything is ok, but we should document it.

Reply:
I agree, better to add a comment here.